src/vm/class.cpp

```diff
@@ -1472,12 +1472,13 @@ CorElementType EEClassLayoutInfo::GetNativeHFATypeRaw()
     CorElementType hfaType = ELEMENT_TYPE_END;
 
 #ifndef DACCESS_COMPILE
-    const FieldMarshaler *pFieldMarshaler = GetFieldMarshalers();
-    while (numReferenceFields--)
+    const NativeFieldDescriptor *pNativeFieldDescriptors = GetNativeFieldDescriptors();
+    for(UINT i = 0; i < numReferenceFields; ++i)
```

Comment:
Instead of this, I would use an iterator style of loop.

```c++
    const NativeFieldDescriptor *curr = GetNativeFieldDescriptors();
    const NativeFieldDescriptor *end = curr + numReferenceFields;
    for (; curr != end; ++curr)
    {
       ...
    }
```